### PR TITLE
Minor improvement to the work approach checklist

### PR DESCRIPTION
**Pull request by Potherca**
- Add utility script to generate a table-of-content
- Replaces `{:toc}` with an actual table of content (so the TOC is visible in the repo as well)
- Moves checklist to separate repo
- Resolves issue raised by @Terrub in previous PR.

### CODE REVIEW

File: bin/add-toc.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#/ replaces all occurrences of "<!-- toc -->" with an actual table of content.
+#/
+#/ Requires the markdown-toc node module to be available in the project.
+# ------------------------------------------------------------------------------
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+shortUsage() {
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: We created a shell script before here with all the guidelines included. The script is here:
https://github.com/DealerDirect/php-qa-tools/blob/master/bin/install.sh

I noticed the following:
- There is no description above this function.

File: bin/add-toc.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            bShowHelp=true
+        fi
+    done
+
+    if [ "${bShowHelp}" == "true" ];then
+        fullUsage
+        shortUsage
+    else
+        add-toc "$@"
+    fi
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: As described above I noticed the following:
- There is no final exit function

File: bin/add-toc.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+add-toc() {
+    local aFiles sDirectory sFile
+     sDirectory="${1}"
+
+    aFiles=$(grep --recursive --files-with-matches --include='*.md' --exclude-dir={node_modules,_site} '<!-- toc -->' "${sDirectory}")
+
+
+    for sFile in "${aFiles[@]}";do
+        ./node_modules/.bin/markdown-toc -i "${sFile}"
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: It is better to use `npm run markdown-toc` or maby even better to use `yarn run markdown-toc` if this is availible in this repository. Even npm uses yarn themselves as you can see here:

https://github.com/nodejs/docker-node/blob/master/7.6/Dockerfile

File: bin/add-toc.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+set -o pipefail
+
+shortUsage() {
+        echo ''
+        echo 'Usage: '
+        echo "         $0 <directory-to-scan>"
+        echo ''
+}
+
+fullUsage() {
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: As described above I noticed the following:
- There is no description above this function.

File: bin/add-toc.sh [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    grep '^#/' <"$0" | cut -c4-
+}
+
+add-toc() {
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: As described above I noticed the following:
- There is no description above this function.

File: bin/add-toc.sh [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    aFiles=$(grep --recursive --files-with-matches --include='*.md' --exclude-dir={node_modules,_site} '<!-- toc -->' "${sDirectory}")
+
+
+    for sFile in "${aFiles[@]}";do
+        ./node_modules/.bin/markdown-toc -i "${sFile}"
+    done
+}
+
+main() {
```
Comment (brammittendorff-dd) [inline bin/add-toc.sh]: As described above I noticed the following:
- There is no description above this function.

**Review by brammittendorff-dd — CHANGES_REQUESTED** [review]

File: bin/add-toc.sh (hunk 1 above)
Comment (Potherca) [reply bin/add-toc.sh]: Will fix.

**Comment by Potherca** [comment]
@brammittendorff-dd I updated the script to address your concerns. Did I miss anything?

**Review by brammittendorff-dd — APPROVED** [review]

**Pull request merged**